sim/shaman/runes.go

```diff
@@ -136,14 +166,20 @@ func (shaman *Shaman) applyPowerSurge() {
 		},
 	)
 
-	var affectedSpells []*core.Spell
+	var spellsAffectedByReset []*core.Spell
+	var spellsAffectedByInstantCast []*core.Spell
 
 	shaman.PowerSurgeAura = shaman.RegisterAura(core.Aura{
 		Label:    "Power Surge Proc",
 		ActionID: core.ActionID{SpellID: 440285},
 		Duration: time.Second * 10,
 		OnInit: func(aura *core.Aura, sim *core.Simulation) {
-			affectedSpells = core.FilterSlice(
+			spellsAffectedByReset = core.FilterSlice(
+				core.Flatten([][]*core.Spell{
+					shaman.ChainLightning,
+					{shaman.LavaBurst},
+				}), func(spell *core.Spell) bool { return spell != nil })
+			spellsAffectedByInstantCast = core.FilterSlice(
```

Comment:
These had to be separated out otherwise trying to reset chain heal threw an error